### PR TITLE
service/dap: Add support for debug and test modes

### PR DESCRIPTION
**Pull request by polinasok**
This change also:
-- refactors building helper functions from commands.go into its own gobuild package.
-- makes processing untyped launch args (that are implementation-specific and not part of the DAP spec) more robust, verified by additional tests

### CODE REVIEW

File: pkg/gobuild/gobuild.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// This can be used to remove the temporary binary generated for the session.
+func Remove(path string) {
+	err := os.Remove(path)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "could not remove %v: %v\n", path, err)
+	}
+}
+
+// OptFlags generates default build flags to turn off optimization and inlining.
+func OptFlags(args []string) []string {
```
Comment (aarzilli) [inline pkg/gobuild/gobuild.go]: As far as I can tell this is only called inside gobuild, it doesn't need to be exported.

Comment (polinasok) [reply pkg/gobuild/gobuild.go]: Good point. Fixed.

File: service/dap/server.go [hunk 2]
```diff
@@ -286,34 +292,71 @@ func (s *Server) onInitializeRequest(request *dap.InitializeRequest) {
...
+			s.sendInternalErrorResponse(request.Seq, err.Error())
+			return
+		}
+
+		buildFlags, ok := request.Arguments["buildFlags"].(string)
+		if !ok {
+			buildFlags = ""
+		}
+
+		if mode == "debug" {
```
Comment (derekparker) [inline service/dap/server.go]: nit: Switch here instead?

Comment (polinasok) [reply service/dap/server.go]: Done

**Pull request merged**